Sources/SourceControl/GitRepository.swift

```diff
@@ -245,6 +258,25 @@ public class GitRepository: Repository, WorkingCheckout {
         }
     }
 
+    public func remove() throws {
+        try queue.sync {
+            // If there is a staged or unstaged diff, don't remove the
+            // tree. This won't detect new untracked files, but it is
+            // just a safety measure for now. This is only valid in a work tree.
+            let isBareRepo = try Git.runPopen([Git.tool, "-C", path.asString, "rev-parse", "--is-bare-repository"]).chomp() == "true"
```

Comment:
... it might make more sense to track what kind of repo we are in explicitly, rather than manually checking this here.